op-node/rollup/driver/driver.go

```diff
@@ -278,10 +278,16 @@ func (s *Driver) eventLoop() {
 
 		planSequencerAction()
 
-		// If the engine is not ready, or if the L2 head is actively changing, then reset the alt-sync:
-		// there is no need to request L2 blocks when we are syncing already.
-		if head := s.SyncDeriver.Engine.UnsafeL2Head(); head != lastUnsafeL2 || !s.SyncDeriver.Derivation.DerivationReady() {
-			s.log.Debug("altSyncTicker reset", "head", head, "lastUnsafeL2", lastUnsafeL2, "derivationReady", s.SyncDeriver.Derivation.DerivationReady())
+		// Reset the alt-sync ticker when either:
+		//   - The unsafe L2 head has changed, or
+		//   - Derivation is not yet ready (unless derivation is intentionally disabled via UnsafeOnly).
+		// This prevents requesting L2 blocks unnecessarily while we are already syncing.
+		//
+		// When UnsafeOnly is enabled, derivation will never become ready. Without the
+		// !UnsafeOnly guard, the alt-sync ticker would reset indefinitely, preventing
+		// normal alt-sync progress.
+		if head := s.SyncDeriver.Engine.UnsafeL2Head(); head != lastUnsafeL2 || (!s.SyncDeriver.Derivation.DerivationReady() && !s.SyncDeriver.SyncCfg.UnsafeOnly) {
+			s.log.Debug("altSyncTicker reset", "head", head, "lastUnsafeL2", lastUnsafeL2, "derivationReady", s.SyncDeriver.Derivation.DerivationReady(), "unsafeOnly", s.SyncDeriver.SyncCfg.UnsafeOnly)
```

Comment:
Rather than one large composite logic statement with comment, this logic can be broken into pieces by first getting the head, then checking each of these cases individually. The appropriate comment about the semantics of the logic can then be attached to each part.

If you're feeling fancy, you could even define the common behavior as a function

```go

helper := func(h head) { //... }

head := s.SyncDeriver.Engine.UnsafeL2Head()


// if the head does not match, set it
if head != lastUnsafeL2 { helper(head) 
// if derivation is enabled but not yet ready, set it
} else if ... { helper(head) }
```

I don't think the logic explanations about what happens if `!UnsafeOnly` is required if we can just break the logic apart a bit.